pymatgen/analysis/graphs.py

```diff
@@ -1002,30 +1023,30 @@ def types_of_coordination_environments(self, anonymous=False):
             connected_sites = self.get_connected_sites(idx)
             connected_species = [connected_site.site.species_string for connected_site in connected_sites]
 
-            labels = []
+            sp_counts = []
             for sp in set(connected_species):
                 count = connected_species.count(sp)
-                labels.append((count, sp))
+                sp_counts.append((count, sp))
 
-            labels = sorted(labels, reverse=True)
+            sp_counts = sorted(sp_counts, reverse=True)
 
             if anonymous:
                 mapping = {centre_sp: "A"}
                 available_letters = [chr(66 + idx) for idx in range(25)]
-                for label in labels:
+                for label in sp_counts:
                     sp = label[1]
                     if sp not in mapping:
                         mapping[sp] = available_letters.pop(0)
                 centre_sp = "A"
-                labels = [(label[0], mapping[label[1]]) for label in labels]
+                sp_counts = [(label[0], mapping[label[1]]) for label in sp_counts]
 
-            labels = [f"{label[1]}({label[0]})" for label in labels]
+            labels = [f"{label[1]}({label[0]})" for label in sp_counts]
             motif = f"{centre_sp}-{','.join(labels)}"
             motifs.add(motif)
 
-        return sorted(motifs)
+        return list(set(motifs))
```

Comment:
Added a `sp_counts` for "species counts" because type changes during runtime and triggered `mypy` error.

However this looks funny, maybe return `set` instead? Would using `set` break anything?